pkg/plugins/builtin/ssh/ssh.go

```diff
@@ -76,6 +79,15 @@ func configssh(i interface{}) error {
 		return fmt.Errorf("ssh too many hops (max %d)", MaxHops)
 	}
 
+	if cfg.Timeout != "" {
+		if cfg.Timeout[0] == '-' {
```

Comment:
This is superfluous, it can be checked after the duration is parsed in a single check.